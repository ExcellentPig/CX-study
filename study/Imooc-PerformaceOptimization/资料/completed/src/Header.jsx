import React from 'react';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';
import {Link} from 'react-router-dom';

const styles = {
    root: {
        flexGrow: 1,
        backgroundColor: "#317EFB"
    },
    brand: {
        flex: 1,
    },
    about: {
        flex: 1,
        textAlign: 'right'
    },
    aboutLink: {
        color: '#f68743',
        textDecoration: 'none',
        border: '1px solid #f68743',
        padding: 6,
        borderTopRightRadius: 10,
        '&:hover': {
            color: '#f3a279',
            border: '1px solid #f3a279',
        },
        '&:active': {
            color: '#fa6a21',
            border: '1px solid #fa6a21',
        }
    }
};

class Header extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            open: false
        };
        this.openDrawer = this.openDrawer.bind(this);
    }
    openDrawer() {
        this.setState({open: true});
    }
    render() {
        return (
            <div className={this.props.classes.root}>
                <AppBar position="static">
                    <Toolbar>
                        <Typography variant="h5" color="inherit" className={this.props.classes.brand}>
                            Dereck's Backyard Sales!
                        </Typography>
                        <Typography variant="h6" color="inherit" className={this.props.classes.about}>
                            <Link className={this.props.classes.aboutLink} to="/about">About</Link>
                        </Typography>
                    </Toolbar>
                </AppBar>
            </div>
        );
    }
}

export default withStyles(styles)(Header);
