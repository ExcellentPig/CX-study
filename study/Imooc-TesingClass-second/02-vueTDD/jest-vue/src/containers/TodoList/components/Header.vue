<template>
  <div class="header">
    <div class="header-content">
      TodoList
      <input placeholder="请输入!" class="header-input" data-test="input" v-model="inputValue" @keyup.enter="addTodoItem">
    </div>
  </div>
</template>

<script>
export default {
  name: 'Header',
  data () {
    return {
      inputValue: ''
    }
  },
  methods: {
    addTodoItem () {
      if (this.inputValue) {
        this.$emit('add', this.inputValue)
        this.inputValue = ''
      }
    }
  }
}
</script>

<style scoped lang="stylus">
.header {
  background-color #666
  height 60px
}
.header-content {
  width: 600px;
  margin 0 auto;
  color #FFF
  font-size 24px
  line-height 60px
}
.header-input {
  float right
  width: 360px;
  line-height 24px
  margin-top 16px
  outline none
  color #333
  text-indent 10px
}
</style>
