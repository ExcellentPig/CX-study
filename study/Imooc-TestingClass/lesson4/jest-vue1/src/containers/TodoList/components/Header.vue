<template>
  <div class="header">
    <div class="header-content">
      TodoList
      <input
      class="header-input"
      data-test="header-input"
      type="text"
      :value="inputValue"
      @input = "e => changeInputValue(e.target.value)"
      @keyup.enter="addTodoItem"
      placeholder="请输入TodoItem"
      >
    </div>
  </div>
</template>

<script>
import { mapState, mapMutations } from 'vuex'
export default {
  name: 'Header',
  props: {},
  // data () {
  //   return {
  //     inputValue: ''
  //   }
  // },
  computed: {
    ...mapState({
      inputValue: state => state.inputValue
    })
  },
  methods: {
    addTodoItem () {
      if (this.inputValue) {
        this.$emit('add', this.inputValue)
        // this.inputValue = ''
        this.changeInputValue('')
      }
    },
    ...mapMutations({
      changeInputValue: 'changeInputValue'
    })
  }
}
</script>

<style scoped lang="stylus">
.header {
  background #333
  line-height 60px
}
.header-content {
  width: 600px;
  margin 0 auto
  color #fff
  font-size 24px
}
.header-input {
  float right
  width: 360px;
  margin: 16px;
  line-height 24px
  outline none
  color #333
  text-indent 10px
}
</style>
