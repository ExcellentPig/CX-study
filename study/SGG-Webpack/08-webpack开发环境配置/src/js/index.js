// 引入 iconfont 样式文件
import '../css/iconfont.css';
import '../css/index.less'


function add (x, y) {
	return x + y
}

console.log(add(1, 2))