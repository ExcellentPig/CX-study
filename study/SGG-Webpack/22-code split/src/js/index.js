function sum (...args) {
	return args.reduce((p,c) => p+c, 0) // 0 是初始化值
}

/**
 * 通过js代码 将某文件打包成单独的chunk
 * import 动态导入语法： 能将某个文件单独打包
 * webpackChunkName: 'test' 设置文件名
 */
import (/* webpackChunkName: 'test' */'./test')
	.then((res) => {
		// 加载成功
		console.log(res)
	})
	.catch(() => {
		console.log('文件加载失败')
	})

console.log(sum(1,2,3,4,5,6))
