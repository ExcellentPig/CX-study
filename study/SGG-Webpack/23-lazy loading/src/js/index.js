console.log('index.js')

// import { mul } from './test.js'

// console.log(mul(3, 4))

document.getElementById('btn').onclick = function () {
	//console.log(mul(4, 4))
	// 懒加载  当文件需要使用时再加载
	// webpackPrefetch: true 预加载  会提前加载js文件  等其他资源加载完毕浏览器空闲了 再偷偷加载资源
	// 正常加载可以认为是并行加载(同一时间加载多个文件)
	import (/* webpackChunkName: 'test', webpackPrefetch: true */'./test').then(({mul}) => {
			console.log(mul(4, 5))
		})
}