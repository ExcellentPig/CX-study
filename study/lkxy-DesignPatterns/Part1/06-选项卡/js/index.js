/*window.onload = function (ev) {
    // 1. 获取需要的标签
    var allLis = $('tab_header').getElementsByTagName('li');
    var allDom = $('tab_content').getElementsByClassName('dom');
    // console.log(allLis, allDom);

    // 2. 遍历监听
    for (var i = 0; i < allLis.length; i++) {
        var li = allLis[i];
        li.id = i;
        li.onmouseover = function () {
            // 2.1 所有的都不被选中
            for(var j=0; j<allLis.length; j++){
                allLis[j].className = '';
                allDom[j].style.display = 'none';
            }
            // 2.2 当前的li被选中
            this.className = 'selected';
            allDom[this.id].style.display = 'block';
        }
    }

};*/

var tab = new TabsFn();
tab.init();

/**
 * 根据id获取标签
 * @param {string}id
 * @returns {object}
 */
function $(id) {
    return typeof id === 'string' ? document.getElementById(id) : null;
}