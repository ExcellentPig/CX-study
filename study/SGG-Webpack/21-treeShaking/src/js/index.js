import { mul } from './test.js'
import '../css/index.css'

function sum (...args) {
	return args.reduce((p,c) => p+c, 0) // 0 是初始化值
}
console.log(mul(2, 4))

console.log(sum(1,2,3,4,5,6))
