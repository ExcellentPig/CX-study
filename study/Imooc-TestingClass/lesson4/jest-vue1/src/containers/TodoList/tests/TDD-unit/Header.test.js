import {
  shallowMount
} from '@vue/test-utils'

import Header from '../../components/Header.vue'
import { findTestWrapper } from '../../../../utils/testUtils.js'

describe('Header组件',()=>{
  it('样式发生改变 做提示',()=>{
    const wrapper = shallowMount(Header)
    expect(wrapper).toMatchSnapshot() //通过快照 当页面有改变时 就会提示快照改变
  })


  it('包含input框',()=>{
    const wrapper = shallowMount(Header)
    const input = findTestWrapper(wrapper,'input')
    expect(input.exists()).toBe(true)
  })

  it('input框的初始内容为空',()=>{
    const wrapper = shallowMount(Header)
    const inputValue = wrapper.vm.inputValue
    expect(inputValue).toBe('')
  })

  it('input框值发生变化，数据应该跟着变',()=>{
    const wrapper = shallowMount(Header)
    const input = findTestWrapper(wrapper,'input')
    input.setValue('dee khan')
    const inputValue = wrapper.vm.inputValue
    expect(inputValue).toBe('dee khan')
  })

  it('input框输入回车，无内容时，无反应',()=>{
    const wrapper = shallowMount(Header)
    const input = findTestWrapper(wrapper,'input')
    input.setValue('')
    input.trigger('keyup.enter')
    expect(wrapper.emitted().add).toBeFalsy()
  })

  it('input框输入回车，有内容时，向外触发事件,同时清空inputValue',()=>{
    const wrapper = shallowMount(Header)
    const input = findTestWrapper(wrapper,'input').at(0)
    input.setValue('dee khan')
    input.trigger('keyup.enter')
    expect(wrapper.emitted().add).toBeTruthy()
    expect(wrapper.vm.inputValue).toBe('')
  })

})
