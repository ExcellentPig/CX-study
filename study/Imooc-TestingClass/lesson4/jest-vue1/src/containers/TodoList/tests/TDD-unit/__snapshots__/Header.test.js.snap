// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header组件 样式发生改变 做提示 1`] = `
<div class="header">
  <div class="header-content">
    TodoList
    <input data-test="input" type="text" placeholder="请输入TodoItem" class="header-input">
  </div>
</div>
`;
